libs/ai-endpoints/langchain_nvidia_ai_endpoints/llm.py

```diff
@@ -0,0 +1,223 @@
+
+    @property
+    def available_models(self) -> List[Model]:
+        """
+        Get a list of available models that work with NVIDIA.
+        """
+        return self._client.get_available_models(self.__class__.__name__)
+
+    @classmethod
+    def get_available_models(
+        cls,
+        **kwargs: Any,
+    ) -> List[Model]:
+        """
+        Get a list of available models that work with the Completions API.
+        """
+        return cls(**kwargs).available_models
+
+    @property
+    def _llm_type(self) -> str:
+        """
+        Get the type of language model used by this chat model.
+        Used for logging purposes only.
+        """
+        return "NVIDIA"
+
+    def _call(
+        self,
+        prompt: str,
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[CallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> str:
+        payload: Dict[str, Any] = {
+            "model": self.model,
+            "prompt": prompt,
+            **self._init_args,
+            **self.__check_kwargs(kwargs),
+        }
+        if stop:
+            payload["stop"] = stop
+
+        if payload.get("stream", False):
+            warnings.warn("stream set to true for non-streaming call, ignoring")
+            del payload["stream"]
+
+        response = self._client.get_req(payload=payload)
+        response.raise_for_status()
+
+        # todo: handle response's usage and system_fingerprint
+
+        choices = response.json()["choices"]
+        # todo: write a test for this by setting n > 1 on the request
+        #       aug 2024: n > 1 is not supported by endpoints
+        if len(choices) > 1:
+            warnings.warn(
+                f"Multiple choices in response, returning only the first: {choices}"
+            )
+
+        return choices[0]["text"]
+
+    def _stream(
+        self,
+        prompt: str,
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[CallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> Iterator[GenerationChunk]:
+        payload: Dict[str, Any] = {
+            "model": self.model,
+            "prompt": prompt,
+            "stream": True,
+            **self._init_args,
+            **self.__check_kwargs(kwargs),
+        }
+        if stop:
+            payload["stop"] = stop
+
+        # we construct payload w/ **kwargs positioned to override stream=True,
+        # this lets us know if a user passed stream=False
+        if not payload.get("stream", True):
+            warnings.warn("stream set to false for streaming call, ignoring")
+            payload["stream"] = True
+
+        for chunk in self._client.get_req_stream(payload=payload):
+            content = chunk["content"]
+            generation = GenerationChunk(text=content)
+            if run_manager:  # todo: add tests for run_manager
+                run_manager.on_llm_new_token(content, chunk=generation)
+            yield generation
```

Comment:
Can we support _identifying_params: https://python.langchain.com/v0.1/docs/modules/model_io/llms/custom_llm/

Reply:
added